pinot-common/src/main/java/com/linkedin/pinot/common/config/AbstractTableConfig.java

```diff
@@ -20,214 +20,183 @@
-  }
-
-  public String getTableType() {
-    return tableType;
+    return _tableName;
   }
 
-  public boolean isRealTime() {
-    return tableType.equalsIgnoreCase(TABLE_TYPE_REALTIME);
+  public TableType getTableType() {
+    return _tableType;
   }
 
   public SegmentsValidationAndRetentionConfig getValidationConfig() {
-    return validationConfig;
+    return _validationConfig;
   }
 
   public TenantConfig getTenantConfig() {
-    return tenantConfig;
+    return _tenantConfig;
   }
 
-  public TableCustomConfig getCustomConfigs() {
-    return customConfigs;
+  public IndexingConfig getIndexingConfig() {
+    return _indexingConfig;
   }
 
-  public abstract IndexingConfig getIndexingConfig();
+  public TableCustomConfig getCustomConfigs() {
+    return _customConfigs;
+  }
 
-  public @Nullable QuotaConfig getQuotaConfig() {
-    return quotaConfig;
+  @Nullable
+  public QuotaConfig getQuotaConfig() {
+    return _quotaConfig;
   }
 
   @Override
   public String toString() {
-    final StringBuilder result = new StringBuilder();
-    result.append("tableName:" + tableName + "\n");
-    result.append("tableType:" + tableType + "\n");
-    result.append("tenant : " + tenantConfig.toString() + " \n");
-    result.append("segments : " + validationConfig.toString() + "\n");
-    result.append("customConfigs : " + customConfigs.toString() + "\n");
-    if (quotaConfig != null) {
-      result.append("quota : " + quotaConfig.toString() + "\n");
+    StringBuilder stringBuilder = new StringBuilder();
+    stringBuilder.append("tableName: ").append(_tableName).append('\n');
+    stringBuilder.append("tableType: ").append(_tableType).append('\n');
+    stringBuilder.append("segmentsConfig: ").append(_validationConfig).append('\n');
+    stringBuilder.append("tenants: ").append(_tenantConfig).append('\n');
+    stringBuilder.append("tableIndexConfig: ").append(_indexingConfig).append('\n');
+    stringBuilder.append("metadata: ").append(_customConfigs).append('\n');
+    if (_quotaConfig != null) {
+      stringBuilder.append("quota: ").append(_quotaConfig).append('\n');
     }
-    return result.toString();
-  }
-
-  public JSONObject toJSON() throws JSONException, JsonGenerationException, JsonMappingException, IOException {
-    JSONObject ret = new JSONObject();
-    ret.put("tableName", tableName);
-    ret.put("tableType", tableType);
-    ret.put("segmentsConfig", new JSONObject(new ObjectMapper().writeValueAsString(validationConfig)));
-    ret.put("tenants", new JSONObject(new ObjectMapper().writeValueAsString(tenantConfig)));
-    ret.put("tableIndexConfig", new JSONObject(new ObjectMapper().writeValueAsString(getIndexingConfig())));
-    ret.put("metadata", new JSONObject(new ObjectMapper().writeValueAsString(customConfigs)));
-    if (quotaConfig != null) {
-      ret.put("quota", new JSONObject(new ObjectMapper().writeValueAsString(quotaConfig)));
+    return stringBuilder.toString();
+  }
+
+  public JSONObject toJSON()
```

Comment:
The AbstractTableConfig is not implemented the way ObjectMapper can serialize/deserialize directly.